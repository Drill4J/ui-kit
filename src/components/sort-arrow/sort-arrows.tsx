import * as React from 'react';
import { BEM } from '@redneckz/react-bem-helper';

import { Icons } from '../icon';

import styles from './sort-arrows.module.scss';

interface Props {
  className?: string;
  onClick?: () => void;
  sort?: 'asc' | 'desc';
}

const sortArrows = BEM(styles);

export const SortArrows = sortArrows(({
  className, onClick, sort,
}: Props) => (
  <div className={className} onClick={onClick}>
    <SortArrowWrapper>
      <SortArrow active={sort === 'asc'} rotate={180} />
      <SortArrow active={sort === 'desc'} />
    </SortArrowWrapper>
  </div>
));

const SortArrowWrapper = sortArrows.sortArrowWrapper('div');
const SortArrow: React.FC<{
  active?: boolean;
  sort?: 'asc' | 'desc';
  rotate?: number;
}> = sortArrows.sortArrow(Icons.SortingArrow);
