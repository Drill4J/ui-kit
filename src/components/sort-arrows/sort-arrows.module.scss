@import '../../style-guide/index';

.sort-arrows {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 24px;
  width: 24px;
  color: map-get($monochrome, medium-tint);
  cursor: pointer;

  &__sort-arrows-wrapper {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 16px;
    width: 16px;
  }

  &__sort-arrow {
    &--active {
      color: map-get($monochrome, default);
    }
  }
}
