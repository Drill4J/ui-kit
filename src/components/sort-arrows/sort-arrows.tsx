import * as React from 'react';
import { BEM, div } from '@redneckz/react-bem-helper';

import { Icons } from '../icon';

import styles from './sort-arrows.module.scss';

interface Props {
  className?: string;
  onClick?: () => void;
  order: 'ASC' | 'DESC' | null;
}

const sortArrows = BEM(styles);

export const SortArrows = sortArrows(({ className, onClick, order }: Props) => (
  <div className={className} onClick={onClick}>
    <SortArrowsWrapper>
      <SortArrow active={order === 'ASC'}>
        <Icons.SortingArrow rotate={180} />
      </SortArrow>
      <SortArrow active={order === 'DESC'}>
        <Icons.SortingArrow />
      </SortArrow>
    </SortArrowsWrapper>
  </div>
));

const SortArrowsWrapper = sortArrows.sortArrowsWrapper('div');
const SortArrow = sortArrows.sortArrow(div({} as { active: boolean }));
