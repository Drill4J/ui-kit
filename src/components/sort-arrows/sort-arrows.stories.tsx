import React from 'react';
import { storiesOf } from '@storybook/react';

import { SortArrows } from './sort-arrows';

storiesOf('SortArrows', module).add('SortArrows', () => {
  const [sort, setSort] = React.useState<'ASC' | 'DESC'>('ASC');
  return <SortArrows order={sort} onClick={() => (sort === 'ASC' ? setSort('DESC') : setSort('ASC'))} />;
});
