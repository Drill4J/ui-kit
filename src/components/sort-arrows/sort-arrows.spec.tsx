import React from 'react';
import renderer from 'react-test-renderer';

import { SortArrows } from './sort-arrows';

describe('SortArrows', () => {
  it('should match snapshot ascending sort-arrows', () => {
    const tree = renderer.create(<SortArrows order="ASC" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot descending sort-arrows', () => {
    const tree = renderer.create(<SortArrows order="DESC" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
