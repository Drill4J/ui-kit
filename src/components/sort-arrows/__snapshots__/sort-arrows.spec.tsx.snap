// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortArrows should match snapshot ascending sort-arrows 1`] = `
<div
  className="sort-arrows"
>
  <div
    className="sort-arrows__sort-arrows-wrapper"
    data-modifiers=""
  >
    <div
      className="sort-arrows__sort-arrow sort-arrows__sort-arrow--active"
    >
      <svg
        className="generic-icon"
        data-modifiers=""
        fillRule="evenodd"
        height={10}
        style={
          Object {
            "WebkitTransform": "rotate(180deg)",
            "opacity": "100%",
          }
        }
        transform="rotate(180)"
        viewBox="0 0 7 10"
        width={7}
      >
        <path
          d="M3.5 10l.01-.001h.033l.03-.006.017-.001.015-.003-.062.01a.5.5 0 0 0 .31-.145l2.95-2.95a.5.5 0 0 0-.707-.707L4 8.295 4 .5a.5.5 0 0 0-1 0l-.001 7.843L.854 6.197a.5.5 0 0 0-.708.707l2.95 2.95a.498.498 0 0 0 .311.144L3.49 10a.51.51 0 0 0 .011 0z"
        />
      </svg>
    </div>
    <div
      className="sort-arrows__sort-arrow"
    >
      <svg
        className="generic-icon"
        data-modifiers=""
        fillRule="evenodd"
        height={10}
        style={
          Object {
            "WebkitTransform": "rotate(0deg)",
            "opacity": "100%",
          }
        }
        transform="rotate(0)"
        viewBox="0 0 7 10"
        width={7}
      >
        <path
          d="M3.5 10l.01-.001h.033l.03-.006.017-.001.015-.003-.062.01a.5.5 0 0 0 .31-.145l2.95-2.95a.5.5 0 0 0-.707-.707L4 8.295 4 .5a.5.5 0 0 0-1 0l-.001 7.843L.854 6.197a.5.5 0 0 0-.708.707l2.95 2.95a.498.498 0 0 0 .311.144L3.49 10a.51.51 0 0 0 .011 0z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`SortArrows should match snapshot descending sort-arrows 1`] = `
<div
  className="sort-arrows"
>
  <div
    className="sort-arrows__sort-arrows-wrapper"
    data-modifiers=""
  >
    <div
      className="sort-arrows__sort-arrow"
    >
      <svg
        className="generic-icon"
        data-modifiers=""
        fillRule="evenodd"
        height={10}
        style={
          Object {
            "WebkitTransform": "rotate(180deg)",
            "opacity": "100%",
          }
        }
        transform="rotate(180)"
        viewBox="0 0 7 10"
        width={7}
      >
        <path
          d="M3.5 10l.01-.001h.033l.03-.006.017-.001.015-.003-.062.01a.5.5 0 0 0 .31-.145l2.95-2.95a.5.5 0 0 0-.707-.707L4 8.295 4 .5a.5.5 0 0 0-1 0l-.001 7.843L.854 6.197a.5.5 0 0 0-.708.707l2.95 2.95a.498.498 0 0 0 .311.144L3.49 10a.51.51 0 0 0 .011 0z"
        />
      </svg>
    </div>
    <div
      className="sort-arrows__sort-arrow sort-arrows__sort-arrow--active"
    >
      <svg
        className="generic-icon"
        data-modifiers=""
        fillRule="evenodd"
        height={10}
        style={
          Object {
            "WebkitTransform": "rotate(0deg)",
            "opacity": "100%",
          }
        }
        transform="rotate(0)"
        viewBox="0 0 7 10"
        width={7}
      >
        <path
          d="M3.5 10l.01-.001h.033l.03-.006.017-.001.015-.003-.062.01a.5.5 0 0 0 .31-.145l2.95-2.95a.5.5 0 0 0-.707-.707L4 8.295 4 .5a.5.5 0 0 0-1 0l-.001 7.843L.854 6.197a.5.5 0 0 0-.708.707l2.95 2.95a.498.498 0 0 0 .311.144L3.49 10a.51.51 0 0 0 .011 0z"
        />
      </svg>
    </div>
  </div>
</div>
`;
